debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -242,7 +297,7 @@ private Schema updateEnvelopeSchema(Schema oldEnvelopeSchema, String newTopicNam
         for (org.apache.kafka.connect.data.Field field : oldEnvelopeSchema.fields()) {
             final String fieldName = field.name();
             Schema fieldSchema = field.schema();
-            if (fieldName == Envelope.FieldName.BEFORE || fieldName == Envelope.FieldName.AFTER) {
+            if (Objects.equals(fieldName, Envelope.FieldName.BEFORE) || Objects.equals(fieldName, Envelope.FieldName.AFTER)) {
```

Comment:
I think object equality actually would have been fine when using this with any of the Debezium connectors, since the Debezium connectors build the `Field` object using the `Envelope`'s actual field name string objects. But I agree this is safer if not more correct.